Categories/Functor/Equivalence.agda

```diff
@@ -1,94 +1,231 @@
+
+  ≡F-identityʳ : {F : Functor C D} → F ∘F idF ≡F F
+  ≡F-identityʳ = record { eq₀ = λ _ → refl ; eq₁ = λ _ → Equiv.sym id-comm }
+
+module _ {C : Category o ℓ e} {D : Category o′ ℓ′ e′}
+         {E : Category o″ ℓ″ e″}
+         where
+
+  open Functor
+  open Category E
+  open HomReasoning hiding (refl; trans)
+  open MorphReasoning E
+  open HId
+  open HIdProps
+  open _≡F_
+
+  ∘F-resp-≡F : {H I : Functor D E} {F G : Functor C D} →
+               H ≡F I → F ≡F G → H ∘F F ≡F I ∘F G
+  ∘F-resp-≡F {H} {I} {F} {G} H≡I F≡G = record
+    { eq₀ = λ X → trans (HF≡HG X) (HG≡IG X)
+    ; eq₁ = λ {X Y} f → begin
+        hid E (trans (HF≡HG Y) (HG≡IG Y)) ∘ F₁ H (F₁ F f)
+      ≈˘⟨ ∘-resp-≈ˡ (hid-trans E _ _) ⟩
+        (hid E (HG≡IG Y) ∘ hid E (HF≡HG Y)) ∘ F₁ H (F₁ F f)
+      ≈˘⟨ ∘-resp-≈ˡ $ ∘-resp-≈ʳ (F-hid H (eq₀ F≡G Y)) ⟩
+        (hid E (HG≡IG Y) ∘ F₁ H (hid D (eq₀ F≡G Y))) ∘ F₁ H (F₁ F f)
+      ≈⟨ glue (eq₁ H≡I (F₁ G f)) ([ H ]-resp-square (eq₁ F≡G f)) ⟩
+        F₁ I (F₁ G f) ∘ (hid E (HG≡IG X) ∘ F₁ H (hid D (eq₀ F≡G X)))
+      ≈⟨ ∘-resp-≈ʳ $ ∘-resp-≈ʳ (F-hid H (eq₀ F≡G X)) ⟩
+        F₁ I (F₁ G f) ∘ (hid E (HG≡IG X) ∘ hid E (HF≡HG X))
+      ≈⟨ ∘-resp-≈ʳ (hid-trans E _ _) ⟩
+        F₁ I (F₁ G f) ∘ hid E (trans (HF≡HG X) (HG≡IG X))
+      ∎
+    }
+    where
+      HF≡HG = λ X → cong (F₀ H) (eq₀ F≡G X)
+      HG≡IG = λ X → eq₀ H≡I (F₀ G X)
+
+module _ {B : Category o ℓ e} {C : Category o′ ℓ′ e′}
+         {D : Category o″ ℓ″ e″} {E : Category o‴ ℓ‴ e‴}
+         where
+
   open Functor
-  open Category C
-  module B = Category B
-  open HomReasoning
-  obj-≡ : (X : Category.Obj A) → F₀ h (F₀ f X) ≡ F₀ i (F₀ g X)
-  obj-≡ X = ≡.trans (cong (F₀ h) (proj₁ f≡g X)) (proj₁ h≡i _)
-  hom-≡ : {a₁ a₂ : Category.Obj A} (a₁⇒a₂ : A [ a₁ , a₂ ]) → subst₂ _⇒_ (obj-≡ a₁) (obj-≡ a₂) (F₁ h (F₁ f a₁⇒a₂)) ≈ F₁ i (F₁ g a₁⇒a₂)
-  hom-≡ {a₁} {a₂} a₁⇒a₂ =
-    let Ff = F₁ f a₁⇒a₂ in
-    let Fg = F₁ g a₁⇒a₂ in
-    let eq₁ X = proj₁ h≡i X in
-    let eq₂ X = proj₁ f≡g X in
-    let eq₃ X = cong (F₀ h) (eq₂ X) in
-    let s = subst₂ _⇒_ (eq₁ _) (eq₁ _) in
-    begin
-    subst₂ _⇒_ (obj-≡ a₁) (obj-≡ a₂) (F₁ h Ff)   ≈˘⟨ ≡⇒≈ $ subst₂-subst₂ _⇒_ (eq₃ _) (eq₁ _) (eq₃ _) (eq₁ _) ⟩
-    s (subst₂ _⇒_ (eq₃ _) (eq₃ _) (F₁ h Ff))     ≈⟨ ≡⇒≈ $ cong s (subst₂-app _⇒_ Ff (λ _ _ x → F₁ h x) (eq₂ _) (eq₂ _)) ⟩
-    s (F₁ h (subst₂ B._⇒_ (eq₂ _) (eq₂ _) Ff))   ≈⟨ subst₂≈ (F-resp-≈ h (proj₂ f≡g a₁⇒a₂)) (eq₁ _) (eq₁ _) ⟩
-    subst₂ _⇒_ (eq₁ _) (eq₁ _) (F₁ h Fg)         ≈⟨  proj₂ h≡i Fg ⟩
-    F₁ i (F₁ g a₁⇒a₂) ∎
+  open Category E
+  open _≡F_
+
+  ≡F-assoc : {F : Functor B C} {G : Functor C D} {H : Functor D E} →
+             (H ∘F G) ∘F F  ≡F  H ∘F (G ∘F F)
+  ≡F-assoc = record { eq₀ = λ _ → refl ; eq₁ = λ _ → Equiv.sym id-comm }
+
+-- For backwards compatibility with the old functor equality.  Should
+-- probably be removed eventually.
+
+module OldEquiv {C : Category o ℓ e} {D : Category o′ ℓ′ e′} where
```

Comment:
Go ahead and delete this entirely, please.

Reply:
Done.